x/autopilot/module_ibc.go

```diff
@@ -123,51 +123,54 @@ func (im IBCModule) OnRecvPacket(
 		return channeltypes.NewErrorAcknowledgement(err)
 	}
 
-	// to be utilized from ibc-go v5.1.0
-	if data.Memo == "stakeibc/LiquidStake" {
-		strideAccAddress, err := sdk.AccAddressFromBech32(data.Receiver)
-		if err != nil {
-			return channeltypes.NewErrorAcknowledgement(err)
-		}
-
-		ack := im.app.OnRecvPacket(ctx, packet, relayer)
-		if ack.Success() {
-			return im.keeper.TryLiquidStaking(ctx, packet, data, &types.ParsedReceiver{
-				ShouldLiquidStake: true,
-				StrideAccAddress:  strideAccAddress,
-			}, ack)
-		}
-		return ack
+	// ibc-go v5 has a Memo field that can store forwarding info
+	// For older version of ibc-go, the data must be stored in the receiver field
+	var metadata string
+	if data.Memo != "" {
+		metadata = data.Memo
+	} else {
+		metadata = data.Receiver
 	}
 
 	// parse out any forwarding info
-	parsedReceiver, err := types.ParseReceiverData(data.Receiver)
+	parsedPacketMetadata, err := types.ParsePacketMetadata(metadata)
```

Comment:
it's a very good question. I think it's a bit tough to dynamically check the flag for each module since the parsing is done once. A couple options I see:
* Add `AtLeastOneRouteActive` helper function to params and call this from module_ibc before any parsing. Only downside here is that we need to remember to modify this function every time we add a parameter.
* Check the `Active` params in the place where each module's metadata is validated. Downside here is that its kinda clunky to pass the params into the parser
* Leave as is. One v minor upside of the current setup is it helps us identify the case where we passed a packet with the memo fields filled out properly - yet the module was inactive